src/semaphore/broadcast/markdown.py

```diff
@@ -0,0 +1,767 @@
+    """The hours of the day (0-23) when the recurrence happens."""
+
+    by_minute: Optional[List[int]] = None
+    """The minutes of the hour (0-23) when the recurrence happens."""
+
+    by_second: Optional[List[int]] = None
+    """The seconds of the minute (0-59) when the recurrence happens."""
+
+    exclude: bool = False
+    """Set to True to exclude these events from the schedule."""
+
+    @validator("timezone", pre=True, allow_reuse=True)
+    def preprocess_timezone(
+        cls, v: Any, values: Dict[str, Any], **kwargs: Any
+    ) -> datetime.tzinfo:
+        """Convert a timezone into a tzinfo instance."""
+        return convert_to_tzinfo(v)
+
+    @validator("date", "start", "end", pre=True, allow_reuse=True)
+    def preprocess_optional_arrow(
+        cls, v: Any, values: Dict[str, Any], **kwargs: Any
+    ) -> Optional[arrow.Arrow]:
+        """Convert a datetime into a arrow.Arrow, or None."""
+        if v is None:
+            return v
+        else:
+            print("converting to arrow")
+            print(v)
+            return convert_to_arrow(
+                v, default_tz=values.get("timezone", dateutil.tz.UTC)
+            )
+
+    @validator("by_set_position", "by_year_day", each_item=True)
+    def check_year_day_index(cls, v: int) -> int:
+        if (v >= 1 and v <= 366) or (v <= -1 and v >= 366):
+            return v
+        else:
+            raise ValueError(
+                "value must be in the range [1, 366] or [-366, -1]"
+            )
+
+    @validator("by_month", each_item=True)
+    def check_month_index(cls, v: int) -> int:
+        if (v >= 1 and v <= 12) or (v <= -1 and v >= -12):
+            return v
+        else:
+            raise ValueError("value must be in the range [1, 12] or [-12, -1]")
+
+    @validator("by_month_day", each_item=True)
+    def check_month_day(cls, v: int) -> int:
+        if (v >= 1 and v <= 31) or (v <= -1 and v >= -31):
+            return v
+        else:
+            raise ValueError("value must be in the range [1, 31] or [-31, -1]")
+
+    @validator("by_week", each_item=True)
+    def check_week(cls, v: int) -> int:
+        if (v >= 1 and v <= 52) or (v <= -1 and v >= -52):
+            return v
+        else:
+            raise ValueError("value must be in the range [1, 52] or [-52, -1]")
+
+    @validator("by_hour", each_item=True)
+    def check_hour(cls, v: int) -> int:
+        if v >= 0 and v <= 23:
+            return v
+        else:
+            raise ValueError("value must be in the range [0, 23]")
+
+    @validator("by_minute", each_item=True)
+    def check_minute(cls, v: int) -> int:
+        if v >= 0 and v <= 59:
+            return v
+        else:
+            raise ValueError("value must be in the range [0, 59]")
+
+    @validator("by_second", each_item=True)
+    def check_second(cls, v: int) -> int:
+        if v >= 0 and v <= 59:
+            return v
+        else:
+            raise ValueError("value must be in the range [0, 59]")
```

Comment:
@rra Do you know how to make type function's like `conint` work inside a List?

Trying:

```
    by_second: Optional[List[conint(ge=0, lt=60)]] = None
    """The seconds of the minute (0-59) when the recurrence happens."""
```

I get:

```
src/semaphore/broadcast/markdown.py:361: error: Invalid type comment or annotation  [valid-type]
src/semaphore/broadcast/markdown.py:361: note: Suggestion: use conint[...] instead of conint(...)
```

Reply:
Oh, yes, I forgot about that. That's going to be a problem, and I don't know how to work around it. Validators for everything it is!